analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -92,35 +92,27 @@ class GoMod(
         val projectDir = definitionFile.parentFile
 
         stashDirectories(projectDir.resolve("vendor")).use {
-            val edges = getDependencyGraph(projectDir)
+            val graph = getDependencyGraph(projectDir)
```

Comment:
commit-message: Was there an issue with the performance of the code? 
If there wasn't I believe the code should rather be optimized in terms of complexity / readability.
Please mentioned that in the commit message.

Reply:
We *do* have some reports about our Go analysis being slow, see e.g. https://github.com/oss-review-toolkit/ort/issues/3860. That also matches my personal experience. I'd say, any performance improvement is welcome in that regard.

Reply:
On top of that it's even slightly less code than before, so 👍🏻 from my side.

Reply:
While I do not have any benchmarks, I noticed that the original code was iterating over the list of edges multiple times. As the number of edges is typically greater than the number of nodes, the new representation should be more efficient, especially as packages and their dependencies can now be looked up directly in the map.